File: JavaScript/119-Pascal's Triangle II.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    let dp = [1];
    for (let i=1; i<=rowIndex; i++){
        let tmp = new Array(i+1).fill(1);
        for (let j=1; j<=i-1; j++){
            tmp[j] = dp[j] + dp[j-1];
        }
        dp = tmp;
    }
    return dp;
};

File: C++/119-Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> curr = vector(1, 1);
        for (int row = 1; row <= rowIndex; row++){
            vector<int> nxt = vector(row+1, 0);
            for (int i=0; i<curr.size(); i++){
                nxt[i] += curr[i];
                nxt[i+1] += curr[i];
            }
            curr = move(nxt); // Use std::move() to transfer ownership of memory, printing nxt[0] will give error
        }
        return curr;
    }
};